crates/ruff_linter/src/message/sarif.rs

```diff
@@ -218,8 +356,11 @@ impl Serialize for SarifResult<'_> {
                 }
             }],
             "ruleId": self.code.as_str(),
-        })
-        .serialize(serializer)
+        });
+        if !self.fixes.is_empty() {
+            value["fixes"] = serde_json::to_value(&self.fixes).unwrap();
+        }
+        value.serialize(serializer)
     }
```

Comment:
Hmm, interesting. Can't we derive `Serialize` on `SarifResult` instead and skip serializing `fixes` if the container is empty (https://serde.rs/field-attrs.html#skip_serializing_if)?